zebra-chain/src/orchard/address.rs

```diff
@@ -25,6 +38,22 @@ impl fmt::Debug for Address {
     }
 }
 
+impl From<Address> for [u8; 43] {
+    /// Corresponds to the _raw encoding_ of an *Orchard* _shielded payment address_.
+    ///
+    /// <https://zips.z.cash/protocol/protocol.pdf#orchardpaymentaddrencoding>
+    fn from(addr: Address) -> [u8; 43] {
+        use std::convert::TryInto;
+
+        let mut bytes = io::Cursor::new(Vec::new());
+
+        let _ = bytes.write_all(&<[u8; 11]>::from(addr.diversifier));
+        let _ = bytes.write_all(&<[u8; 32]>::from(addr.transmission_key));
+
+        bytes.into_inner().try_into().expect("43 bytes")
```

Comment:
Optional ideas:

1. It's possible to avoid the `Vec` heap allocation and resizing using a mutable slice borrow.
2. I think it might be a good idea to use `expect` here to detect any issues if we accidentally change something in the future.

```suggestion
        let mut bytes = [0u8; 43];
        let mut writer = &mut bytes[..];

        writer
            .write_all(&<[u8; 11]>::from(addr.diversifier))
            .expect("diversifier should fit in address buffer");

        writer
            .write_all(&<[u8; 32]>::from(addr.transmission_key))
            .expect("transmission should fit in address buffer");

        bytes
```

Reply:
I had an idea for an alternative that doesn't need the runtime checks. It could be simpler if the `address_bytes` and `address_end` bindings are removed, but I was worried about having the number `11` repeated and having the risk of forgetting to update one of the occurrences in the future :sweat_smile: 

```suggestion
        let mut bytes = [0u8; 43];

        let address_bytes: [u8; 11] = addr.diversifier.into();
        let address_end = address_bytes.len();

        bytes[..address_end].copy_from_slice(&address_bytes);
        bytes[address_end..].copy_from_slice(&<[u8; 32]>::from(addr.transmission_key));

        bytes
```